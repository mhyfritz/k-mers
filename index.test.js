/* global test expect */

const kmers = require('.')

test('3-mers of "foobar" via `all`', () => {
  const mers = kmers(3, 'foobar').all()
  expect(mers).toMatchSnapshot()
})

test('3-mers of "foobar" via `next`', () => {
  const mers = []
  const kmersIter = kmers(3, 'foobar')
  while (true) {
    const kmer = kmersIter.next()
    if (kmer.value === undefined) {
      break
    }
    mers.push(kmer)
  }
  expect(mers).toMatchSnapshot()
})

test('2nd and 3rd 3-mer of "foobar" via `seek`', () => {
  const kmersIter = kmers(3, 'foobar')
  const mers = []
  kmersIter.seek(1)
  mers.push(kmersIter.next())
  mers.push(kmersIter.next())
  expect(mers).toMatchSnapshot()
})

test('1-mers of "foobar" via `all`', () => {
  const mers = kmers(1, 'foobar').all()
  expect(mers).toMatchSnapshot()
})

test('6-mers of "foobar" via `all`', () => {
  const mers = kmers(6, 'foobar').all()
  expect(mers).toMatchSnapshot()
})
