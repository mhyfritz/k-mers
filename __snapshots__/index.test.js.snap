// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1-mers of "foobar" via \`all\` 1`] = `
Array [
  "f",
  "o",
  "o",
  "b",
  "a",
  "r",
]
`;

exports[`2nd and 3rd 3-mer of "foobar" via \`seek\` 1`] = `
Array [
  Object {
    "index": 1,
    "value": "oob",
  },
  Object {
    "index": 2,
    "value": "oba",
  },
]
`;

exports[`3-mers of "foobar" via \`all\` 1`] = `
Array [
  "foo",
  "oob",
  "oba",
  "bar",
]
`;

exports[`3-mers of "foobar" via \`next\` 1`] = `
Array [
  Object {
    "index": 0,
    "value": "foo",
  },
  Object {
    "index": 1,
    "value": "oob",
  },
  Object {
    "index": 2,
    "value": "oba",
  },
  Object {
    "index": 3,
    "value": "bar",
  },
]
`;

exports[`6-mers of "foobar" via \`all\` 1`] = `
Array [
  "foobar",
]
`;
